### PR TITLE
invalidate PrefixData cache on rm_rf for conda-build

### PR DESCRIPTION
**Pull request by kalefranz**

### CODE REVIEW

File: conda/exports.py [hunk 1]
```diff
@@ -181,3 +181,12 @@ def __call__(self, *args, **kw):
...
                 return value
+
+
+from .gateways.disk.delete import rm_rf as _rm_rf  # NOQA
+from .core.linked_data import PrefixData as _PrefixData  # NOQA
+
+
+def rm_rf(path, max_retries=5, trash=True):
+    _rm_rf(path, max_retries, trash)
+    _PrefixData._cache_ = {}
```
Comment (mingwandroid) [inline conda/exports.py]: Do we really need to kill all of the cache here?

**Comment by kalefranz** [comment]
That's what @msarahan asked for yesterday when we spoke about it. At least that was my understanding.

**Comment by msarahan** [comment]
I don't think you need to kill the entire cache.  You definitely need to nuke any entry for ``path`` here, but if there are any other paths in the cache, they can (and should) stay.

**Comment by kalefranz** [comment]
Cool I'll make that change then.

**Pull request merged**